openlibrary/plugins/upstream/tests/test_utils.py

```diff
@@ -102,3 +102,22 @@ def test_canonical_url():
     web.ctx.query = '?sort=new&mode=2'
     url = 'https://www.openlibrary.org/authors/Ayn_Rand'
     assert request.canonical_url == url
+
+
+def test_reformat_html():
+    f = utils.reformat_html
+
+    input_string = '<p>This sentence has 32 characters.</p>'
+    assert f(input_string, 10) == 'This sente...'
+    assert f(input_string) == 'This sentence has 32 characters.'
+    assert f(input_string, 5000) == 'This sentence has 32 characters.'
+
+    multi_line_string = """<p>This sentence has 32 characters.</p>
+                           <p>This new sentence has 36 characters.</p>"""
+    assert f(multi_line_string) == 'This sentence has 32 ' \
+        'characters.<br>This new sentence has 36 characters.'
+    assert f(multi_line_string, 34) == 'This sentence has 32 ' \
+        'characters.<br>T...'
+
+    assert f("<script>alert('hello')</script>", 34) == "alert('hello')"
+    assert f("&lt;script&gt;") == "&lt;script&gt;"
```

Comment:
I added a test case which is failing; this should return `&lt;script&gt;`, but return `<script>` 😬 Is there a way we can make it preserve things like `&gt;`?

Reply:
Just pushed an update that handles entity references.  Also updated the failing test case to satisfy previously undocumented precondition for HTML strings.

Reply:
The update didn't work properly.  Will our markdown converter output entity references?  If I add `&gt;` to a list description, it appears as `>` in the HTML.

Reply:
Yeah, if the user write `&lt;`, we'll get `&lt;` out ; the HTML is hard to read from the browser, cause it changes it to be more user-readable.